backend/src/stwo/prover.rs

```diff
@@ -18,15 +35,102 @@ impl<F: FieldElement> StwoProver<F> {
     #[allow(dead_code)]
     #[allow(unused_variables)]
     pub fn new(
-        _analyzed: Arc<Analyzed<F>>,
+        analyzed: Arc<Analyzed<F>>,
         _fixed: Arc<Vec<(String, Vec<F>)>>,
-        setup: Option<&mut dyn io::Read>,
+        //  setup: Option<&mut dyn io::Read>,
     ) -> Result<Self, io::Error> {
         Ok(Self {
-            _analyzed,
+            analyzed,
             _fixed,
             _proving_key: None,
             _verifying_key: None,
         })
     }
+    pub fn prove(
+        &self,
+        witness: &[(String, Vec<F>)],
+        witgen_callback: WitgenCallback<F>,
+    ) -> Result<Vec<u8>, String> {
+        let config = PcsConfig {
+            pow_bits: 16,                          // Any value you want to set for pow_bits
+            fri_config: FriConfig::new(0, 1, 100), // Using different numbers for FriConfig
+        };
+
+        // Precompute twiddles.
+        let twiddles = SimdBackend::precompute_twiddles(
+            CanonicCoset::new(
+                (self.analyzed.degree() as u32) + 1 + config.fri_config.log_blowup_factor,
+            )
+            .circle_domain()
+            .half_coset,
+        );
+
+        // Setup protocol.
+        let prover_channel = &mut Poseidon252Channel::default();
+        let commitment_scheme =
+            &mut CommitmentSchemeProver::<SimdBackend, Poseidon252MerkleChannel>::new(
+                config, &twiddles,
+            );
+
+        let trace = PowdrCircuit::new(self.analyzed.clone())
+            .with_witgen_callback(witgen_callback)
+            .with_witness(witness)
+            .into_stwo_circuit_trace();
+
+        let mut tree_builder = commitment_scheme.tree_builder();
+        tree_builder.extend_evals(trace);
+        tree_builder.commit(prover_channel);
+
+        let component = PowdrComponent::new(
+            &mut TraceLocationAllocator::default(),
+            PowdrEval::new(
+                self.analyzed.clone(),
+                self.analyzed.commitment_count()
+                    + self.analyzed.constant_count()
+                    + self.analyzed.publics_count(),
+            ),
+        );
+
+        //let start = Instant::now();
+        let proof = stwo_prover::core::prover::prove::<SimdBackend, Poseidon252MerkleChannel>(
```

Comment:
Same, to `config.rs`, we can define our own `Proof` type there so that we don't need to pass generics here.